Change Name -> Title, resolve conflicts with Base

Ayushk4 opened the pull request:
- [X] CodeBase
- [X] Docs
- [X] Tests

I have made the following changes in the relevant places.
- `name` -> `title`
- `name!` -> `title!`
- `names` -> `titles`
- `names!` -> `titles!`

Earlier the `name` ( now `title`) was set by default to `Unnamed Document`, should that also be changed to `Untitled Document`?

laschuet commented:
> Earlier the `name` ( now `title`) was set by default to `Unnamed Document`, should that also be changed to `Untitled Document`?

I would like that change.

Ayushk4 commented:
@laschuet I have changed it :slightly_smiling_face:. I added it as a separate commit. In case someone may want it to be reverted, we may go back by one commit.
